### PR TITLE
add linux build

### PR DESCRIPTION
**Pull request by sudopluto**
based off of the wonderful work done by @bernhardkaindl in  https://github.com/TheCherno/Walnut/pull/11.

will be paired with corresponding pull req in WalnutApp

### CODE REVIEW

**Comment by bernhardkaindl** [comment]
@jamesthomasgriffin @TheCherno Please merge to let Linux users join the fun!

File: Walnut/src/Walnut/EntryPoint.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #pragma once
 
-#ifdef WL_PLATFORM_WINDOWS
+#if defined WL_PLATFORM_WINDOWS || defined WL_PLATFORM_LINUX
```
Comment (DaveAxiom) [inline Walnut/src/Walnut/EntryPoint.h:3]: This doesn't address the Windows only code below in the WL_DIST block.

File: WalnutExternal.lua [hunk 2]
```diff
@@ -3,20 +3,62 @@
...
+   if errorCode ~= 0 then
+      premake.error("Install " .. package .. " using package manager")
+   end
+   premake.info(package .. " is installed")
+end
 
 group "Dependencies"
    include "vendor/imgui"
-   include "vendor/glfw"
+   if os.ishost("linux") and os.istarget("linux") then
```
Comment (DaveAxiom) [inline WalnutExternal.lua:30]: The filter keyword seems to be best practice for setting OS specific settings.
`filter "system:linux"`

**Comment by DaveAxiom** [comment]
This Linux PR branch doesn't work like #11 because it doesn't specify links in WalnutApp! See `WalnutApp/premake5.lua` in #11.

The Microsoft C++ compiler links at object compile time. All other compilers link at the generation of the executable in the linking stage.

File: WalnutExternal.lua [hunk 3]
```diff
@@ -3,20 +3,62 @@
...
 
 LibraryDir = {}
-LibraryDir["VulkanSDK"] = "%{VULKAN_SDK}/Lib"
+LibraryDir["VulkanSDK"] = "%{VULKAN_SDK}/lib"
 
 Library = {}
-Library["Vulkan"] = "%{LibraryDir.VulkanSDK}/vulkan-1.lib"
+
+if os.ishost("linux") and os.istarget("linux") then
+   Library["Vulkan"] = "%{LibraryDir.VulkanSDK}/libvulkan.so.1"
```
Comment (DaveAxiom) [inline WalnutExternal.lua:15]: All that is needed is `links { "vulkan" }` for the compiler to link in the Vulkan library. If it were the case that the Vulkan library were installed in a non-standard directory, only then would it be required to specify that additional directory.

Comment (bernhardkaindl) [reply WalnutExternal.lua:15]: The best what you can do is to apply your changes on your tree, commit them, push the new branch to your clone of Walnut on github and open a new merge request, referencing #22 and #11.

I don't have any capacity currently to continue working on it, just commit your changes and open a new pull request.

**Pull request closed without merging**